holidays/countries/guinea.py

```diff
@@ -0,0 +1,186 @@
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC
+from holidays.groups import ChristianHolidays, InternationalHolidays, IslamicHolidays
+from holidays.holiday_base import HolidayBase
+from holidays.observed_holiday_base import ObservedHolidayBase, SUN_TO_NEXT_MON
+
+
+class Guinea(
+    ObservedHolidayBase, HolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays
+):
+    """Guinea holidays.
+
+    References:
+        * [Decree No. 2022-0526](https://igt.gov.gn/wp-content/uploads/2024/10/D-2022-0526-PRG-CNRD_221103_131021.pdf)
+        * <https://www.timeanddate.com/holidays/guinea/>
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Guinea>
+        * <https://anydayguide.com/calendar/1878>
+        * <https://www.timeanddate.com/holidays/guinea/second-republic-day>
+        * <https://www.timeanddate.com/holidays/guinea/all-saints-day>
+
+    Note:
+        According to Decree No. 2022-0526(Article 2) of 2 November 2022, if New Year's Day,
+        Independence Day or Eid al-Fitr fall on a non-working day, the next working day is
+        also a Holiday.
+        Eid al-Adha was assigned two holidays from 2022(Decree No. 2022-0526(Article 1))
+    """
+
+    country = "GN"
+    default_language = "fr"
+    # %s (estimated).
+    estimated_label = tr("%s (estimé)")
+    # Day after the %s.
+    observed_label = tr("Lendemain de la %s")
+    # Day after the %s (estimated).
+    observed_estimated_label = tr("Lendemain de la %s (estimé)")
+    supported_languages = ("en_US", "fr")
+
+    # Guinea gained independence from France on October 2, 1958
+    start_year = 1959
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=GuineaIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        kwargs.setdefault("observed_rule", SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        dt = self._add_new_years_day(tr("Fête du Nouvel an"))
+        if self._year >= 2023:
+            self._add_observed(dt)
+
+        if self._year <= 2021:
```

Comment:
```suggestion
        if 1985 <= self._year <= 2021:
```
Since the Second Republic itself began in 1983. The test should also be corrected accordingly.

Reply:
No, why should it?

Reply:
> The test should also be corrected accordingly.
❗

Reply:
@nalin-28, you need to update test.